### PR TITLE
Register audit/v1beta1 for master config

### LINKED ISSUE #18752: audit config doesn't accept v1beta1
You get an error about needing to be v1alpha1.

/assign @soltysh

### PR DESCRIPTION
**Pull request by soltysh**
Fixes #18752.

/assign @deads2k 
/cherry-pick 3.9

### CODE REVIEW

**Comment by soltysh** [comment]
This and https://github.com/openshift/origin/pull/18751 are needed in 3.9

File: pkg/cmd/server/apis/config/validation/master.go [hunk 1]
```diff
@@ -242,7 +242,7 @@ func ValidateAuditConfig(config configapi.AuditConfig, fldPath *field.Path) Vali
 		} else {
 			policyConfiguration, ok := config.PolicyConfiguration.(*auditinternal.Policy)
 			if !ok {
-				validationResults.AddErrors(field.Invalid(fldPath.Child("policyConfiguration"), config.PolicyConfiguration, "must be of type audit/v1alpha1.Policy"))
```
Comment (deads2k) [inline pkg/cmd/server/apis/config/validation/master.go]: message is wrong.  it should allow both.

**Comment by deads2k** [comment]
Once bitten deserves a test I'd think.  That would also ensure my other fixes don't accidentally regress.  test-integration will let you wire it to a file and to an embedded.

File: pkg/cmd/server/apis/config/validation/master.go (hunk 1 above)
Comment (soltysh) [reply pkg/cmd/server/apis/config/validation/master.go]: If you have audit/v1alpha1 it won't complain. If you don't I'd prefer you use v1beta1, that's why I went with that.

**Comment by soltysh** [comment]
Yeah, I agree with the test, will add one on Monday.

**Comment by soltysh** [comment]
@deads2k I've added the tests you've asked for, ptal

**Comment by deads2k** [comment]
Doesn't work, but 

/lgtm

As long as the tests stay.

**Comment by soltysh** [comment]
> As long as the tests stay.

I had to change that embed tests to use internal types only, not sure why or how but it works just fine for me locally, but failed (which I'd expect) on our CI.

**Pull request merged**